pkg/controller/controller.go

```diff
@@ -509,6 +509,26 @@ func (p *csiProvisioner) Provision(options controller.ProvisionOptions) (*v1.Per
 		return nil, capErr
 	}
 
+	if needSnapshotSupport {
+		contentSource := rep.GetVolume().ContentSource
+		if contentSource == nil {
+			sourceErr := fmt.Errorf("created volume contentsource missing")
+			delReq := &csi.DeleteVolumeRequest{
+				VolumeId: rep.GetVolume().GetVolumeId(),
+			}
+			delReq.Secrets = provisionerCredentials
+			ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
+			defer cancel()
+			_, err := p.csiClient.DeleteVolume(ctx, delReq)
+			if err != nil {
+				sourceErr = fmt.Errorf("%v. Cleanup of volume %s failed, volume is orphaned: %v", sourceErr, pvName, err)
```

Comment:
Adding in the original sourceErr from above, I was worried that might get masked out but you covered it.